drake/solvers/NloptSolver.cpp

```diff
@@ -0,0 +1,269 @@
+
+#include <iostream>
+#include <stdexcept>
+#include <list>
+#include <vector>
+
+#include <nlopt.hpp>
+
+#include "NloptSolver.h"
+#include "Optimization.h"
+
+namespace Drake {
+namespace {
+Eigen::VectorXd MakeEigenVector(const std::vector<double>& x) {
```

Comment:
Can this function be made private?  Are there other functions in this file that can be private?